### ISSUE TITLE
Notes

### ISSUE THREAD

**Issue by AntonC9018** [GATAKOLA/Labs-Valik#1]
https://github.com/GATAKOLA/Labs-Valik/blob/a764b4ca44bed1a99bc5ef89bafde29c2d868fc4/Lab1/Lab1.cpp#L8-L10

> Определите массив из 3 интов, заполните нулями.

---

**Comment by AntonC9018** [comment]
https://github.com/GATAKOLA/Labs-Valik/blob/a764b4ca44bed1a99bc5ef89bafde29c2d868fc4/Lab2/example.cpp

Смени encoding на UTF-8.

---

https://github.com/GATAKOLA/Labs-Valik/blob/a764b4ca44bed1a99bc5ef89bafde29c2d868fc4/Lab3/lab3.cpp#L24-L25

Это не все типы и объекты. Повтори синтаксис определения поля и про объекты. Еще одна подсказка это то, что все переменные - это объекты (кроме ссылок), но не все объекты - переменные.

---

https://github.com/GATAKOLA/Labs-Valik/blob/a764b4ca44bed1a99bc5ef89bafde29c2d868fc4/Lab4/Lab4-%D0%A1%D0%BB%D0%BE%D0%B6%D0%BD%D0%BE%D0%B5%20%D0%BB%D0%BE%D0%B3%D0%B8%D1%87%D0%B5%D1%81%D0%BA%D0%BE%D0%B5%20%D0%B2%D1%8B%D1%80%D0%B0%D0%B6%D0%B5%D0%BD%D0%B8%D0%B5.cpp#L23-L34

Тут идея была только инициализацию для `bool result` оставить в блоке. Сам `result` использовать вне блока. Так инициализация этой переменной была бы как бы отдельно вынесена.

---

.gitignore создай, удали все exe файлы из гита через `git rm --cached`.

---

https://github.com/GATAKOLA/Labs-Valik/blob/a764b4ca44bed1a99bc5ef89bafde29c2d868fc4/Lab5/Lab5-%D0%97%D0%B0%D0%B4%D0%B0%D0%BD%D0%B8%D0%B52.cpp#L5-L21

`a` и `b` назови по нормальному. encoding смени.

Понимаешь как итераторы работают?

**Comment by GATAKOLA** [comment]
Я добавил изменения согласно тому что вы попросили. На счёт 3 лабы, надеюсь это все типы и объекты. Как работают итераторы я в целом понимаю - это объект который позваляет перебирать элементы контейнеров и получать доступ к ним. Есть разные варианты итераторов: ввода и вывода(только читают либо записывают данные), последовательный (имеет возможности предыдущего, а так же можно создавать итератор по умолчанию, разыменовывать элемент и инкрементировать итератор), двунаправленный(меет возможности предыдущего, а так же позваляет перемещаться по контейнеру от начала к концу и наоборот) и итераторы произвольного доступа(совмещают в себе все функции предыдущих, а так же определение растояния между итераторами и опрации упорядочивания).

**Comment by AntonC9018** [comment]
> Есть разные варианты итераторов: ввода и вывода(только читают либо записывают данные), последовательный (имеет возможности предыдущего, а так же можно создавать итератор по умолчанию, разыменовывать элемент и инкрементировать итератор), двунаправленный(меет возможности предыдущего, а так же позваляет перемещаться по контейнеру от начала к концу и наоборот) и итераторы произвольного доступа(совмещают в себе все функции предыдущих, а так же определение растояния между итераторами и опрации упорядочивания).

Это загуглил щас?) Таких подробностей никто не спрашивал)

**Comment by GATAKOLA** [comment]
Нет, прочитал на мудле: https://moodle.usm.md/mod/lesson/view.php?id=268336&pageid=2844

**Comment by GATAKOLA** [comment]
С остальным всё ок? Или надо ещё что-то изменить?

**Comment by AntonC9018** [comment]
побеседуем на паре после теста

**Comment by GATAKOLA** [comment]
Я только щас увидел это сообщение на почте, и уже никак. Можете здесь отписать что да как?

**Comment by AntonC9018** [comment]
Давай позже мит сделаем

**Comment by GATAKOLA** [comment]
хорошо, без проблемм. Напишете тогда во сколько и когда(если не сегодня), чтоб я не мониторил постоянно гит

**Comment by AntonC9018** [comment]
https://github.com/GATAKOLA/Labs-Valik/commit/386726e26ff3e15a573c8e7ecc735fcf43ed80f6

**Comment by GATAKOLA** [comment]
Внёс изменения: 
https://github.com/GATAKOLA/Labs-Valik/commit/fe9001ae6b380d2d8a218dbf70b008cf10b96a76
https://github.com/GATAKOLA/Labs-Valik/commit/2bb24e76577289a06ba197b4da710b0d894a6ae3
https://github.com/GATAKOLA/Labs-Valik/commit/35bd9057cfa37d81b848ed8ea5d6c98179632acd
Добавил новые лабы:
https://github.com/GATAKOLA/Labs-Valik/commit/d4c2cb2df9892f1862569fc6d517ab433600a92c
https://github.com/GATAKOLA/Labs-Valik/commit/f1270e83c0458d36c77923cd5e676b446daeabc2
